### PR TITLE
ardupilot: added MAV_CMD_SET_HAGL

### PR DESCRIPTION
**Pull request by tridge**
allows for height above ground from external sources

### CODE REVIEW

File: message_definitions/v1.0/ardupilotmega.xml [hunk 1]
```diff
@@ -340,6 +340,16 @@
         <param index="6" label="Longitude">Longitude</param>
         <param index="7" label="Altitude" units="m">Altitude, not used. Should be sent as NaN. May be supported in a future version of this message.</param>
       </entry>
+      <entry value="43004" name="MAV_CMD_SET_HAGL" hasLocation="false" isDestination="false">
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml:343]: Needs to be 43005 to not conflict with the wind command that we haven't brought back from upstream (but should).

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: message_definitions/v1.0/ardupilotmega.xml (hunk 1 above)
Comment (peterbarker) [reply message_definitions/v1.0/ardupilotmega.xml:343]: https://github.com/ArduPilot/mavlink/pull/363

Comment (tridge) [reply message_definitions/v1.0/ardupilotmega.xml]: changed, and comment on 43004 added

**Review by peterbarker — APPROVED** [review]
Good once format is fixed

**Comment by suraqis** [comment]
Wasn't it better to use 43003 with NaN lat long and not NaN altitude? And we have spare field param4 for altitude accuracy there

**Pull request merged**